sky/clouds/gcp.py

```diff
@@ -625,8 +627,18 @@ def _list_reservations_for_instance_type_in_zone(
         reservations = self._list_reservations_for_instance_type(instance_type)
         return [r for r in reservations if r.zone.endswith(f'/{zone}')]
 
+    def _get_or_create_ttl_cache(self):
+        if getattr(self, '_list_reservations_cache', None) is None:
```

Comment:
May want to add `# Clusters created older than #2352 has GCP objects serialized without the attribute. So we access it this way.`